ui/app/components/console/ui-panel.js

```diff
@@ -0,0 +1,96 @@
+import Ember from 'ember';
+import {
+  parseCommand,
+  extractDataAndFlags,
+  logFromResponse,
+  logFromError,
+  logErrorFromInput,
+} from 'vault/lib/console-helpers';
+
+const { inject, computed } = Ember;
+
+export default Ember.Component.extend({
+  classNames: 'console-ui-panel-scoller',
+  classNameBindings: ['isFullscreen:fullscreen'],
+  isFullscreen: false,
+  console: inject.service(),
+  inputValue: null,
+  log: computed.alias('console.log'),
+
+  logAndOutput(command, logContent) {
+    this.set('inputValue', '');
+    this.get('console').logAndOutput(command, logContent);
+  },
+
+  executeCommand(command, shouldThrow = false) {
+    let service = this.get('console');
+    let serviceArgs;
+    if (command === 'clearall') {
+      this.logAndOutput(command);
+      service.clearLog(true);
+      return;
+    }
+    if (command === 'clear') {
+      this.logAndOutput(command);
+      service.clearLog();
+      return;
+    }
+    if (command === 'fullscreen') {
+      this.toggleProperty('isFullscreen');
+      this.logAndOutput(command);
+      return;
+    }
```

Comment:
This method is really long and I initially found it a little hard to follow. Some random thoughts on how to simplify or break it up:

- Breaking out these special-cased commands into a separate method; or
- Using a data structure to map commands to actions, unifying the UI and back-end commands; or
- Splitting up the method into parse/validate/apply functions with clearly defined interfaces.

Not essential!